rfcs/20240311-gather-scatter-batching-dims.md

```diff
@@ -0,0 +1,427 @@
+| (I4)  | `collapsed_slice_dims`        | 1-dimensional tensor constant of type `si64` | (C1), (C6-C9), (C22)                       |
+| (I5)  | `operand_batching_dims`       | 1-dimensional tensor constant of type `si64` | (C1), (C6), (C10-C12), (C16-C18), (C22)    |
+| (I6)  | `start_indices_batching_dims` | 1-dimensional tensor constant of type `si64` | (C13-C17)                                  |
+| (I7)  | `start_index_map`             | 1-dimensional tensor constant of type `si64` | (C3), (C18-C19)                            |
+| (I8)  | `index_vector_dim`            | constant of type `si64`                      | (C2-C3), (C15), (C22)                      |
+| (I9)  | `slice_sizes`                 | 1-dimensional tensor constant of type `si64` | (C9), (C12), (C20-C22)                     |
+| (I10) | `indices_are_sorted`          | constant of type `i1`                        |                                            |
+
+#### Outputs
+
+| Name     | Type                                  | Constraints     |
+|----------|---------------------------------------|-----------------|
+| `result` | tensor or per-tensor quantized tensor | (C5), (C22-C23) |
+
+#### Constraints
+
+* (C1) `rank(operand) = size(offset_dims) + size(collapsed_slice_dims) + size(operand_batching_dims)`.
+* (C2) `0 <= index_vector_dim <= rank(start_indices)`.
+* (C3) `size(start_index_map) =
+       index_vector_dim < rank(start_indices) ?
+       dim(start_indices, index_vector_dim) : 1`.
+* (C4) `is_unique(offset_dims) and is_sorted(offset_dims)`.
+* (C5) `0 <= offset_dims < rank(result)`.
+* (C6) `is_unique(collapsed_slice_dims + operand_batching_dims)`
+* (C7) `is_sorted(collapsed_slice_dims)`.
+* (C8) `0 <= collapsed_slice_dims < rank(operand)`.
+* (C9) `slice_sizes[collapsed_slice_dims...] <= 1`.
+* (C10) `is_sorted(operand_batching_dims)`.
+* (C11) `0 <= operand_batching_dims < rank(operand)`.
+* (C12) `slice_sizes[operand_batching_dims...] <= 1`.
+* (C13) `is_unique(start_indices_batching_dims) and is_sorted(start_indices_batching_dims)`.
+* (C14) `0 <= start_indices_batching_dims < rank(start_indices)`.
+* (C15) `index_vector_dim not in start_indices_batching_dims`.
+* (C16) `size(operand_batching_dims) == size(start_indices_batching_dims)`.
+* (C17) `dim(operand, operand_batching_dims...) = dim(start_indices, start_indices_batching_dims...)`.
+* (C18) `is_unique(start_index_map + operand_batching_dims)`.
+* (C19) `0 <= start_index_map < rank(operand)`.
+* (C20) `size(slice_sizes) = rank(operand)`.
+* (C21) `0 <= slice_sizes <= shape(operand)`.
+* (C22) `shape(result) = combine(batch_dim_sizes, offset_dim_sizes)` where:
+  * `batch_dim_sizes = shape(start_indices)` except that the dimension size
+    of `start_indices` corresponding to `index_vector_dim` is not included.
+  * `offset_dim_sizes = shape(slice_sizes)` except that the dimension sizes
```

Comment:
It seems that `offset_dim_sizes = shape(slice_sizes) except ...` should be `offset_dim_sizes = slice_sizes except ...`.  Though, this error is inherited from the current wording and is not specific to this RFC.

Reply:
Done.

Reply:
It seems that the merge conflicts are not resolved?

Reply:
Sorry about that!